pyflakes/checker.py

```diff
@@ -1292,13 +1319,16 @@ def TUPLE(self, node):
 
     def IMPORT(self, node):
         for alias in node.names:
+            name = alias.name
             if '.' in alias.name and not alias.asname:
-                importation = SubmoduleImportation(alias.name, node)
+                importation = SubmoduleImportation(name, node)
             else:
                 name = alias.asname or alias.name
                 importation = Importation(name, node, alias.name)
             self.addBinding(node, importation)
 
+            self.store_global_scope(name, alias)
```

Comment:
This needs to be `self.store_global_scope(importation.name, alias)` , otherwise the following will fail:

```py
        self.flakes('''
            def f(): global foo; import foo.bar
            def g(): foo.is_used()
        ''')
```

You can add that test into `test_assignedToGlobal`.

Importing `foo.bar` is a bit strange -- it imports and binds the module `foo` as that name, and then imports `foo.bar` but there is no additional binding in the scope for `foo.bar`.

The determination of the binding name is done within `SubmoduleImportation`.

Note it would be much better if you did the `self.store_global_scope(...)` inside `addBinding`

Reply:
Sorry for responding too late, @jayvdb.
I was reading the `addBinding` and there is something I can't understand.
Where does `scope` comes from? I can't find it...

Anyway, Thanks for reviewing :)

Reply:
self-answering...

if break statement is occured, the variable in for loop can be used at outside of it.